Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/BoundsControl.cs

```diff
@@ -0,0 +1,1155 @@
+                            scaleFactor = clampedScale[0] / initialScaleOnGrabStart[0];
+                        }
+                    }
+
+                    Target.transform.localScale = clampedScale;
+                    Target.transform.position = initialPositionOnGrabStart * scaleFactor + (1 - scaleFactor) * oppositeCorner;
+                }
+            }
+        }
+
+        #endregion Private Methods
+
+
+        #region Used Event Handlers
+
+        void IMixedRealityFocusChangedHandler.OnFocusChanged(FocusEventData eventData)
+        {
+            if (eventData.NewFocusedObject == null)
+            {
+                proximityEffect.ResetProximityScale();
+            }
+
+            if (activation == BoundsControlActivationType.ActivateManually || activation == BoundsControlActivationType.ActivateOnStart)
+            {
+                return;
+            }
+
+            if (!DoesActivationMatchFocus(eventData))
+            {
+                return;
+            }
+
+            bool handInProximity = eventData.NewFocusedObject != null && eventData.NewFocusedObject.transform.IsChildOf(transform);
+            if (handInProximity == wireframeOnly)
+            {
+                wireframeOnly = !handInProximity;
+                // todo: move this out?
+                ResetVisuals();
+            }
+        }
+
+        void IMixedRealityFocusHandler.OnFocusExit(FocusEventData eventData)
+        {
+            if (currentPointer != null && eventData.Pointer == currentPointer)
+            {
+                DropController();
+            }
+        }
+
+        void IMixedRealityFocusHandler.OnFocusEnter(FocusEventData eventData) { }
+
+        private void OnPointerUp(MixedRealityPointerEventData eventData)
+        {
+            if (currentPointer != null && eventData.Pointer == currentPointer)
+            {
+                DropController();
+                eventData.Use();
+            }
+        }
+
+        private void OnPointerDown(MixedRealityPointerEventData eventData)
+        {
+            if (currentPointer == null && !eventData.used)
+            {
+                GameObject grabbedHandle = eventData.Pointer.Result.CurrentPointerTarget;
+                Transform grabbedHandleTransform = grabbedHandle.transform;
+                currentHandleType = GetHandleType(grabbedHandleTransform);
+                if (currentHandleType != HandleType.None)
+                {
+                    currentPointer = eventData.Pointer;
+                    initialGrabPoint = currentPointer.Result.Details.Point;
+                    currentGrabPoint = initialGrabPoint;
+                    initialScaleOnGrabStart = Target.transform.localScale;
+                    initialPositionOnGrabStart = Target.transform.position;
+                    grabPointInPointer = Quaternion.Inverse(eventData.Pointer.Rotation) * (initialGrabPoint - currentPointer.Position);
+
+                    // todo: move this out?
```

Comment:
Either remove or file a bug. It will increase likelihood we complete the work if we have bugs for our todos.

Reply:
i would like to tackle these todos in a follow up PR. I left them there on purpose. I didn't want to move more things around in this PR because it was already getting too big.